src/components/molecules/Visualizer/Engine/Cesium/cameraLimiter.ts

```diff
@@ -0,0 +1,241 @@
+import {
+  Cartesian3,
+  Cartographic,
+  Color,
+  EllipsoidGeodesic,
+  PolylineDashMaterialProperty,
+  Math,
+  Camera as CesiumCamera,
+  Rectangle,
+} from "cesium";
+import type { Viewer as CesiumViewer } from "cesium";
+import { useEffect, useMemo, useState, RefObject } from "react";
+import { CesiumComponentRef } from "resium";
+
+import { Camera } from "@reearth/util/value";
+
+import type { SceneProperty } from "..";
+
+import { getCamera } from "./common";
+
+const targetWidth = 1000000;
+const targetLength = 1000000;
+
+export function useCameraLimiter(
+  cesium: RefObject<CesiumComponentRef<CesiumViewer>>,
+  camera: Camera | undefined,
+  property: SceneProperty["cameraLimiter"] | undefined,
+) {
+  const geodesic = useMemo(():
+    | undefined
+    | { geodesicVertical: EllipsoidGeodesic; geodesicHorizontal: EllipsoidGeodesic } => {
+    const viewer = cesium.current?.cesiumElement;
+    if (!viewer || viewer.isDestroyed()) return undefined;
+    return property?.cameraLimitterTargetArea?.lng && property.cameraLimitterTargetArea.lat
+      ? getGeodesic(
+          viewer,
+          property.cameraLimitterTargetArea.lng,
+          property.cameraLimitterTargetArea.lat,
+        )
+      : undefined;
+  }, [cesium, property?.cameraLimitterTargetArea?.lng, property?.cameraLimitterTargetArea?.lat]);
+
+  // calculate inner limiter dimensions
+  const limiterDimensions = useMemo((): InnerLimiterDimensions | undefined => {
+    if (!geodesic) return undefined;
+
+    const width = property?.cameraLimitterTargetWidth || targetWidth;
+    const length = property?.cameraLimitterTargetLength || targetLength;
```

Comment:
```suggestion
    const width = typeof property?.cameraLimitterTargetWidth === "number" ?  property.cameraLimitterTargetWidth : targetWidth;
    const length = typeof property?.cameraLimitterTargetLength === "number" ? property.cameraLimitterTargetLength : targetLength;
```

Reply:
@rot1024 , I thought there is no need for this, as the min value defined in manifest is `5` so this case won't happen.